src/ast/nodes/shared/FunctionBase.ts

```diff
@@ -121,6 +121,9 @@ export default abstract class FunctionBase extends NodeBase {
 			return this.getObjectEntity().hasEffectsOnInteractionAtPath(path, interaction, context);
 		}
 		if (this.async) {
+			if (this.annotationNoSideEffects) {
+				return false;
+			}
```

Comment:
Yeah, making sense. What do you think about this approach?

Reply:
That would also work, but you can also skip the last part about "parameter side effects". This is basically parameters that have side effects in their default values, or destructuring with computed variables where the computation has side effects. With our annotation, I would also expect those side effects to be ignored. So it is really only the very first part about non-call-related side effects that I would want to keep.